[blog] launch (#26)
gitpod-io/website: src/blog/gitpod-launch.md

---
subtitle: No Setups, Less Waiting, More Flow.

> title: Introducing Gitpod - Frictionless Coding on GitHub

Reviewer: ` – ` (`alt` + `-`)

Author: sorry, I don't get this comment :-)

Reviewer: AFAIK that should be a dash ([en dash actually](https://www.smashingmagazine.com/2012/05/applying-macrotypography-for-readable-web-page/)) and not a hyphen.

Author: I didn't even know there's a difference.

date: Mon Apr 5 2019 15:00:00 GMT+0000 (UTC)
image: https://cdn-images-1.medium.com/max/2000/1*GszIo6iudvQRcmZcBfVmIA.gif
author: svenefftinge
---

Every day developers waste millions of hours switching contexts between projects and branches, setting up development environments or simply waiting for builds to complete.

This friction is not only time consuming but is a serious barrier for contributors. Reading through a long setup description and messing up my own laptop is just too much effort.

Today, we are happy to announce the launch of [Gitpod.io](https://gitpod.io), an online service that provides disposable, ready-to-code development environments for GitHub projects.

## One Click Dev Environments for GitHub

> The idea is simple: You are on GitHub and want to start working on some project. Instead of going through the manual setups, etc. you simply click a button. A ready-to-code dev environment is started in your browser. You can now code, review or just try out the code. Once you are done you simply close the tab.

Reviewer: `or just try the app` instead of `or just try out the code` to avoid the repetition, WDYT?

Author: project?

Reviewer: 👍

![GitHub button](./gitpod-launch/browser-ext.png)

You don’t see the button on GitHub yet? You need to [install a small browser extension](/docs/20_Browser_Extension) to have it. Alternatively, you can just prefix any GitHub URL with “https://gitpod.io/#”.

You can try Gitpod right now. It's free for public repositories.

## No Setups

> Not all projects are equal. We maintain a [developer friendly docker image](https://github.com/gitpod-io/workspace-images/blob/master/full/Dockerfile) that includes all the commonly used tools and version managers. But you can bring your own docker image by adding a __.gitpod.yml__ file to the repository root. Therein you point to any docker image or file and Gitpod will pick it up and even build the image if needed. Read [Gero's post](/blog/docker-in-gitpod) for more details.

Reviewer 2: `image or file and Gitpod will` -> `image or file, and Gitpod will`

> Not all projects are equal. We maintain a [developer friendly docker image](https://github.com/gitpod-io/workspace-images/blob/master/full/Dockerfile) that includes all the commonly used tools and version managers. But you can bring your own docker image by adding a __.gitpod.yml__ file to the repository root. Therein you point to any docker image or file and Gitpod will pick it up and even build the image if needed. Read [Gero's post](/blog/docker-in-gitpod) for more details.

Reviewer: > But you can bring your own docker image by adding a __.gitpod.yml__ file to the repository root. Therein you point to any docker image or file

Maybe a bit shorter? Like this:
But you can point to your own docker image in a __.gitpod.yml__ and Gitpod will ...

To optimize the experience further, the __.gitpod.yml__ accepts [further configuration](/docs/40_Configuration), where you can automatically start builds, watcher processes and dev servers in multiple terminals. You can configure how they open in the workbench and make sure the right ports are exposed and previews open automatically.

As a project owner you should do everything you can to streamline the experience for contributors, so everybody including yourself can focus on being creative and writing code. Gitpod lets you automate the setup and make sure developers get beamed into a dev environment that really is ready to code.

A great example is [ssloy's tinyraytracer project](https://github.com/ssloy/tinyraytracer/wiki). It's a tutorial for raytracing in C++ and is configured so it runs the build and automatically opens the rendered graphics. The author has even added Gitpod buttons to the individual steps in the tutorial to make it easier following along.

![Tinyraytracer](./gitpod-launch/tinyrt.png)

## Less Waiting
There is an additional bit of friction, that every developer has to go through regularly:

> Running the build and downloading dependencies.

Author: Yes, but is it important to correct? Technically, the build includes downloading dependencies. I add it to emphasize it.

> Running the build and downloading dependencies.

Reviewer 3: Not very important. This was more like reversing the order of "More Flow, Less Waiting, No Setup" in the website.

Starting today, [the new Gitpod app is available on the GitHub marketplace](https://github.com/marketplace/gitpod-io). It is the first of its kind and the only one in the new IDE category.

Once you've installed the app on your GitHub repository, it will pre-build dev environments on every commit. So when anybody opens a Gitpod environment on your project, the dev environment is opened as described above. But now also the build ran through and all dependencies are already there.

We call this feature prebuilt workspaces and you can read more about it in [Chris' post](/blog/prebuilt-workspaces).

![Prebuilt Workspaces](./gitpod-launch/prebuilt-workspaces.png)

## More Flow

> The dev environments open differently, depending on the GitHub page you are coming from. You can, for instance, open branches or files by simply going to the repective GitHub page and press the button (or prefix) there. 

Reviewer 2: `repective` -> `respective`

When coming from an issue, Gitpod understands you want to fix it, so you’ll get a fresh local branch and the commit message is populated with a reference to the issue.

> Starting a workspace from a pull request will open in code-review mode. The list of changes are opened on the left so you can easily go through the changes which open in diff editors on the right. Of course you can comment within the IDE and even submit your code review.

Reviewer 2: `The list of changes are opened` -> `The list of changes is opened`

Reviewer 2: `Of course you` -> `Of course, you`

Even if you prefer your local machine for development, being able to do [deep code reviews](/blog/when-code-reviews-lgtm) in a browser is a great way to improve your workflow.

![Inlined Code Review](./gitpod-launch/inline-comments.png)

## Not Just a Playground
While Gitpod keeps the convenience and accessibility of online coding playgrounds and sandboxes, it is not a toy but a complete dev environment running in the cloud. 

> We have partnered with Google, Ericsson, Arm and Red Hat to develop the [open-source project Eclipse Theia](https://theia-ide.org). The project puts [VS Code](https://code.visualstudio.com/)’s state-of-the art technology and UX into the browser based on a sustainable and extendable architecture.

Reviewer 2: `state-of-the art` -> `state-of-the-art`

Reviewer 4: I don't think we can use the word "partner" here. It has [a definition in the legal context](https://www.entrepreneur.com/encyclopedia/partnership) which is not fulfilled. 

We could instead say "we teamed up with Google, Ericsson, Arm..."

> In addition to a the best code editor, Gitpod environments provide full featured terminals to Linux containers running in the cloud. As a result, you get the exact same command line tools you would use locally together with the hottest IDE tech available today. And all of that from within a browser tab.

Reviewer 2: `addition to a the best` -> `addition to the best`

> In addition to a the best code editor, Gitpod environments provide full featured terminals to Linux containers running in the cloud. As a result, you get the exact same command line tools you would use locally together with the hottest IDE tech available today. And all of that from within a browser tab.

Reviewer 3: Typo, safer language, wording:
```suggestion
In addition to one of the best code editors, Gitpod environments provide full featured terminals backed by Linux containers running in the cloud. As a result, you get the same command line tools you would use locally together with the hottest IDE tech available today. And all of that in a browser tab.
```

## VS Code Extensions

> Gitpod will very soon allow installing VS Code extensions directly. Some of the language support is already fueled by VS Code extensions. Support in Theia is at roughly 95% so we don’t yet open up that possibility to users as we want to harden things further. But Gitpod already comes with VS Code extensions pre-installed. For instance, the Go support comes from Microsoft’s very decent extension (https://github.com/Microsoft/vscode-go). We will be adding more popular VS Code extensions shortly before we are opening up the possibility for users to define which extensions they want.

Reviewer 4: ~harden~ -> stabilize. 

Harden: make something more secure; make something more difficult to hack or break

> Gitpod will very soon allow installing VS Code extensions directly. Some of the language support is already fueled by VS Code extensions. Support in Theia is at roughly 95% so we don’t yet open up that possibility to users as we want to harden things further. But Gitpod already comes with VS Code extensions pre-installed. For instance, the Go support comes from Microsoft’s very decent extension (https://github.com/Microsoft/vscode-go). We will be adding more popular VS Code extensions shortly before we are opening up the possibility for users to define which extensions they want.

Reviewer 3: - Clarify "directly [from the marketplace]"
- Wording: "fueled by" --> "powered by"
- Add "with [several] VS Code extensions pre-installed"
- Simplify sentence after "shortly"

```suggestion
Gitpod will very soon allow installing VS Code extensions directly from the marketplace. Some of the language support is already powered by VS Code extensions. Support in Theia is at roughly 95% so we don’t yet open up that possibility to users as we want to harden things further, but Gitpod already comes with several VS Code extensions pre-installed. For instance, the Go support comes from Microsoft’s very decent extension (https://github.com/Microsoft/vscode-go). We will be adding more popular VS Code extensions shortly, and later allow users to define any extensions they want.
```

## Collaboration
Gitpod comes with two major features supporting collaboration.
 - [Sharing running dev environments](/docs/33_Sharing_and_Collaboration) with other users comes in handy if you want to look at code with a colleague to hunt down a bug together or do some pair programming. Collaborators will see the same code you see and even the terminals are synced and shared.

>  - [Snapshots](/blog/workspace-snapshots) are useful in many ways. In a nutshell, they allow for taking snapshots of dev environments at any point in time. You get a link that you can share wherever you want. Users opening such a snapshot link will get an exact clone of the environment including all state and even UI layout. This is great for sharing reproducibles in issues, but also good for blog posts or solutions on Stackoverflow.

Reviewer 2: [typo]: `reproducibles` -> `reproducible`

Reviewer 4: ~are useful in many ways. In a nutshell, they allow for taking snapshots of dev environments at any point in time. You get a link that you can share wherever you want~

better:
> are frozen copies of your workspace. You can create them at any time and you'll get a link you can share wherever you want.

Author: > [typo]: reproducibles -> reproducible

@kittaakos I think it should be plural or do you mean something else?

>  - [Snapshots](/blog/workspace-snapshots) are useful in many ways. In a nutshell, they allow for taking snapshots of dev environments at any point in time. You get a link that you can share wherever you want. Users opening such a snapshot link will get an exact clone of the environment including all state and even UI layout. This is great for sharing reproducibles in issues, but also good for blog posts or solutions on Stackoverflow.

Reviewer 3: Typos:
```suggestion
 - [Snapshots](/blog/workspace-snapshots) are useful in many ways. In a nutshell, they allow taking snapshots of dev environments at any point in time. You get a link that you can share wherever you want. Users opening such a snapshot link will get an exact clone of the environment including all state and even UI layout. This is great for sharing reproducibles in issues, but also good for blog posts or solutions on Stack Overflow.
```

Reviewer 3: Maybe mention "but without any credentials"? Not sure.

![Collaboration](/gitpod-launch/collaboration.png)

## Summary

> Gitpod removes tedious plumbing tasks and reduces painful waiting time. We no longer need to maintain and sync our dev environments but rely on a reproducible setup. All you need is a browser.

Reviewer 3: Wording:
```suggestion
Gitpod removes tedious plumbing tasks and reduces painful waiting time. We no longer need to maintain and sync our dev environments manually but can rely on a reproducible setup. All you need is a browser.
```

> We are looking forward to seeing how this improves contributor experience on open-source projects, simplifies workshops & hackathons and just in general lets developers spend more time being creative and in the flow.

Reviewer 3: ```suggestion
We are looking forward to seeing how this improves contributor experiences on open-source projects, simplifies workshops & hackathons and just in general lets developers spend more time being creative and in the flow.
```

Gitpod is free for open-source and available now.

## Thank You ❤️

> I want to take the opportunity to thank all the people who have helped making Gitpod a reality. All the enthusiastic early adopters providing great feedback during the beta phase, contributors who have helped identifying bugs and sharing their use cases and of course the Eclipse Theia community working with passion on an open-source IDE platform for the future.

Reviewer 2: `making` -> `to make`

Reviewer 2: `identifying` -> `to identify`

> I want to take the opportunity to thank all the people who have helped making Gitpod a reality. All the enthusiastic early adopters providing great feedback during the beta phase, contributors who have helped identifying bugs and sharing their use cases and of course the Eclipse Theia community working with passion on an open-source IDE platform for the future.

Reviewer 3: Wording:
```suggestion
I want to take the opportunity to thank all the people who have helped making Gitpod a reality. All the enthusiastic early adopters who provided great feedback during the beta phase, contributors who have helped identifying bugs and sharing their use cases and of course the Eclipse Theia community working with passion on an open-source IDE platform for the future.
```

> Finally, of course, I want to send a big shout out to the TypeFox team. You are absolutely awesome and being able to work with you on such a great product is everything I want.

Reviewer 3: Aww 😊 same here!
